### PR TITLE
Fixing Queried Content Items query validation (Lombiq Technologies: OCORE-209)

### LINKED ISSUE #17102: Deployment with queries content items step can't be created
### Describe the bug
Deployment with queries content items step can't be created

### Orchard Core version
 v2.1.2

### To Reproduce
1. Go to 'Configurations'
2. Click on 'Import/Export'
3. Scroll down to 'Deployment Plans'
4. Create a deployment plan and manage steps.
5. Add "Queried Content Items"
6. Choose a query and click "create"
7. Then you can see the error.

### Logs and screenshots
image
image
image
The "query.CanReturnContentItems" is always "false".

### PR DESCRIPTION
**Pull request by Piedone**
Fixes #17102. The code was originally introduced in https://github.com/OrchardCMS/OrchardCore/pull/16402. Seems like a no-brainer fix but perhaps I'm not thinking of some gotcha.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Queries/Deployment/QueryBasedContentDeploymentSource.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ protected override async Task ProcessAsync(QueryBasedContentDeploymentStep step,
             return;
         }
 
-        if (!query.CanReturnContentItems || !query.ReturnContentItems)
+        if (!query.CanReturnContentItems && !query.ReturnContentItems)
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Queries/Deployment/QueryBasedContentDeploymentSource.cs:29]: Do we know what is the difference between the CanReturnContentItems and ReturnContentItems?

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.Queries/Deployment/QueryBasedContentDeploymentSource.cs]: `CanReturnContentItems` seems to me like an abandoned view model property. It's not actually used anywhere, I've removed it.

**Review by MikeAlhayek — APPROVED** [review]
I am assuming you tested the changes with and without the checkbox.

**Comment by Piedone** [comment]
Yep, and also exported, then imported, all works.

**Pull request merged**